AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -42,20 +42,31 @@ @implementation _ASCollectionViewCell
 - (void)setNode:(ASCellNode *)node
 {
   _node = node;
-  node.selected = self.selected;
-  node.highlighted = self.highlighted;
+  if (node.selected != self.selected) {
+    [node __setSelectedFromUIKit:self.selected];
```

Comment:
Cool, this new internal wrapper method helps simplify down some of the complex handling here around the interaction delegate (since we don't want internally-generated calls like this to trigger informing UIKit of changes)>

I think you can remove the additional != if around these calls now that the latest version of the diff includes these inside the __set methods.